docs/dev-environment-setup.md

```diff
@@ -0,0 +1,233 @@
+- Clone a local copy of [bcgov/cas-ciip-portal](https://github.com/bcgov/cas-ciip-portal).
+- This project uses Git submodules to incorporate other CAS-maintained repositories. Configuring a [post-checkout Git hook](https://ttboj.wordpress.com/2014/05/06/keeping-git-submodules-in-sync-with-your-branches/) in your local repository can help smooth the experience of using submodules.
+- From the project root, run `git submodule init` - one time only - to initialize the submodules, then `git submodule update` to clone them and checkout the appropriate state based on the specifications in [`.gitmodules`](https://github.com/bcgov/cas-ciip-portal/blob/develop/.gitmodules).
+
+## 2. Package Management
+
+[asdf](https://asdf-vm.com/#/) is a universal package manager used in this project to install and manage versions of the tools listed in [`.tool-versions`](https://github.com/bcgov/cas-ciip-portal/blob/develop/.tool-versions). To do this, several asdf plugins will be installed as well.
+
+- [Install asdf](https://asdf-vm.com/#/core-manage-asdf) using your system's package manager
+- From the project root, run `make install_asdf_tools`, which installs the [additional asdf plugins](https://asdf-vm.com/#/plugins-all) needed to manage the various tools in `.tool-versions`.
+  * this script also [imports the Node.js release team's OpenPGP keys](https://xscode.com/asdf-vm/asdf-nodejs) for checksum verification.
+  * the script also attempts to install Postgres `--with-libxml` via asdf to include a database plugin for XML. Your system will need `libxml2` installed for this.
+  * lastly, it installs `pip` dependencies that relate to `pre-commit`, which helps us run linters to keep things tidy.
+  * you can verify pre-commit works by running: `pre-commit run --all-files` from the project root.
+- You may need to troubleshoot individual steps in `make install_asdf_tools` on your system; in that case, find this target in the root `Makefile` and run the steps individually.
+- `asdf reshim` should be run after asdf installations to update symlinks for the installed packages.
+- Set the version of Postgres installed by asdf [as the global version](https://asdf-vm.com/#/core-manage-versions?id=set-current-version), necessary to prevent later problems installing Sqitch, our database migration tool.
+- `psql --version` should verify the installed version of Postgres.
+- [Yarn](https://yarnpkg.com/) is used as the Javascript package manager, and was installed in this step by asdf from the `.tool-versions`. All `yarn` commands must be run from the same directory as the `package.json` (within `app/`).
+
+**If you're on Linux**, you can now skip to [Step 4](#4-set-up-a-working-perl-5-environment).
+
+<!-- If using a BCGov Macbook, see: https://apps.nrs.gov.bc.ca/int/confluence/display/DEVGUILD/One-time+Setup? -->
```

Comment:
I've left this in a comment for now until we can verify: After reading through the link from @pbastia and others, it doesn't seem like there is anything novel here that couldn't be deduced from following the [asdf documentation](https://asdf-vm.com/#/core-manage-asdf?id=install)?

If someone thinks it's helpful and can describe the link in a sentence, we can uncomment it.

Reply:
I think what I'm trying to get at is, how does this advice relate specifically to a Macbook provided by BCGov and would not relate to any other device?

Reply:
I think you're right, this is just standard asdf install. I don't think there is any need for this anymore